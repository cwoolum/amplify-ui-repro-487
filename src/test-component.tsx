import { Button, Flex } from "@aws-amplify/ui-react";

export function TestComponent() {
  return (
    <Flex>
      I'm some text
      <Flex>
        <Button>Click me</Button>
      </Flex>
    </Flex>
  );
}
