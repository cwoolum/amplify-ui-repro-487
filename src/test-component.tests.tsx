import { mount, render } from "enzyme";
import toJson from "enzyme-to-json";
import { TestComponent } from "test-component";

describe("TestComponent", () => {
  it("should match the snapshot", () => {
    const wrapper = render(<TestComponent></TestComponent>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("find stuff", () => {
    const wrapper = mount(<TestComponent></TestComponent>);

    console.log(wrapper.debug());

    expect(wrapper.find("Button")).toHaveLength(1);
  });
});
