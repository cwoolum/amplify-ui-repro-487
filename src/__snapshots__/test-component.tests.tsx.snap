// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should match the snapshot 1`] = `
<div
  class="amplify-flex"
>
  I'm some text
  <div
    class="amplify-flex"
  >
    <button
      class="amplify-button amplify-field-group__control"
      data-fullwidth="false"
      type="button"
    >
      Click me
    </button>
  </div>
</div>
`;
